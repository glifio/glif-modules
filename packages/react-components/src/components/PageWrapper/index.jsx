import styled from 'styled-components'

export default styled.div`
  display: flex;
  flex-wrap: wrap;
  /* Temp implementation to simplistically handle large scale displays. This should be removed and a more dynamic solution introduced e.g https://css-tricks.com/optimizing-large-scale-displays/  */
  max-width: ${props => props.theme.sizes[props.theme.sizes.length - 1]}px;
  margin: 0 auto;
  min-height: 100vh;
`
