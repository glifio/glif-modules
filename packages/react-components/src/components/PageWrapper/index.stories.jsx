import theme from '../theme'
import ThemeProvider from '../ThemeProvider'

import PageWrapper from './index'

export default {
  title: 'PageWrapper/PageWrapper',
  component: PageWrapper,
  decorators: [
    Story => (
      <div style={{ display: 'flex', justifyContent: 'center' }}>
        <ThemeProvider theme={theme}>{Story()}</ThemeProvider>
      </div>
    )
  ],
  parameters: { actions: { argTypesRegex: '^on.*' } }
}

const Template = args => <PageWrapper {...args} />

export const Base = Template.bind({})
Base.args = {
  children: <div>Hello world</div>,
  bg: 'blue'
}
