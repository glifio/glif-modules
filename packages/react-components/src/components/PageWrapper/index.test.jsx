import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import theme from '../theme'
import ThemeProvider from '../ThemeProvider'

import PageWrapper from '.'

describe('PageWrapper', () => {
  afterEach(cleanup)
  test('renders the PageWrapper', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <PageWrapper>Test</PageWrapper>
      </ThemeProvider>
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
