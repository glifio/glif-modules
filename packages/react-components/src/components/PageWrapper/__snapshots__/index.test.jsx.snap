// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageWrapper renders the PageWrapper 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  max-width: 1440px;
  margin: 0 auto;
  min-height: 100vh;
}

<div
  class="c0"
>
  Test
</div>
`;
